pyam/netcdf.py

```diff
@@ -0,0 +1,90 @@
+import numpy as np
+import datetime as dt
+import pandas as pd
+try:
+    import xarray as xr
+except ImportError:
+    import pip
+    pip.main(['install', '--user', 'xarray'])
+    import xarray as xr
+    
+from pyam.core import IamDataFrame
+from pyam.utils import META_IDX, IAMC_IDX
+
+def read_netcdf(path):
+    """Read timeseries or year-based data and meta-indicators from aggregated netCDF
+    (intermediate from Euro Calliope netCDF)
```

Comment:
Good practice to keep the first sentence of a long docstring to one line. Also, this function should be generic, not limited to Calliope.
```suggestion
    """Read timeseries or year-based data and meta-indicators from netCDF
```

Reply:
I think you'll struggle to write a generic reader if you don't have an equivalent writer (more complete thoughts here: https://github.com/IAMconsortium/pyam/pull/879#issuecomment-2470035042)